src/main/java/anticope/rejects/modules/BonemealAura.java

```diff
@@ -62,12 +68,37 @@ private BlockPos getCrop() {
                         if (age < cropBlock.getMaxAge())
                             return blockPos;
                     }
+                    if (block instanceof CocoaBlock) {
+                        int age = mc.world.getBlockState(blockPos).get(CocoaBlock.AGE);
+                        if (age < 2)
+                            return blockPos;
+                    }
+                    if (block instanceof StemBlock) {
+                        int age = mc.world.getBlockState(blockPos).get(StemBlock.AGE);
+                        if (age < StemBlock.MAX_AGE)
+                            return blockPos;
+                    }
+                    if (block instanceof MushroomPlantBlock) {
+                            return blockPos;
+                    }
+                    if (block instanceof SweetBerryBushBlock) {
+                        int age = mc.world.getBlockState(blockPos).get(SweetBerryBushBlock.AGE);
+                        if (age < 3)
+                            return blockPos;
+                    }
+                     if (block instanceof SaplingBlock || block instanceof AzaleaBlock){
+                            return blockPos;
+                    }
                 }
             }
         }
         return null;
     }
 
+    private boolean canPlaceSapling(BlockPos blockPos) {
```

Comment:
what's the purpose of this function?

Reply:
Sorry abiut that, in theory it should check if tree can grown or not. I have to delete it